From fa4ce27e4d29298bfd5a50d49e85571c2f724623 Mon Sep 17 00:00:00 2001
From: Semih Yavuz <semih.yavuz@qt.io>
Date: Wed, 1 Oct 2025 10:43:12 +0200
Subject: [PATCH 176/239] qmlhighlighting: wrap semantic highlighting logic in
 a namespace

Introduce QmlHighlighting namespace for better organization. Move
utility functions to nested Utils namespace. Update all references
throughout the codebase accordingly.

Pick-to: 6.8
Task-number: QTBUG-140645
Change-Id: I3ad4e5fdd4abdfcb746441acf29594711af13f90
Reviewed-by: Sami Shalayel <sami.shalayel@qt.io>
(cherry picked from commit e8f9576d9350c2af12d582c44f0cb8c4e160198c)
---
 src/qmlls/qqmlhighlightsupport.cpp            | 27 +++++++-------
 src/qmlls/qqmlhighlightsupport_p.h            | 12 +++----
 src/qmlls/qqmlsemantictokens.cpp              | 36 ++++++++++---------
 src/qmlls/qqmlsemantictokens_p.h              | 20 +++++------
 .../auto/qmlls/modules/tst_qmlls_modules.cpp  | 16 +++++----
 .../qmlls/utils/tst_qmlls_highlighting.cpp    | 15 ++++----
 6 files changed, 66 insertions(+), 60 deletions(-)

diff --git a/src/qmlls/qqmlhighlightsupport.cpp b/src/qmlls/qqmlhighlightsupport.cpp
index 8b7c85ad62..60d9e6f8cd 100644
--- a/src/qmlls/qqmlhighlightsupport.cpp
+++ b/src/qmlls/qqmlhighlightsupport.cpp
@@ -37,7 +37,7 @@ QList<QByteArray> defaultTokenModifiersList()
 
 QList<QByteArray> extendedTokenTypesList()
 {
-    return enumToByteArray<HighlightingUtils::SemanticTokenProtocolTypes>();
+    return enumToByteArray<QmlHighlighting::SemanticTokenProtocolTypes>();
 }
 
 /*!
@@ -47,7 +47,7 @@ https://microsoft.github.io/language-server-protocol/specifications/specificatio
 Sends a QLspSpecification::SemanticTokens data as response that is generated for the entire file.
 */
 SemanticTokenFullHandler::SemanticTokenFullHandler(QmlLsp::QQmlCodeModel *codeModel)
-    : QQmlBaseModule(codeModel), m_mode(HighlightingUtils::HighlightingMode::Default)
+    : QQmlBaseModule(codeModel), m_mode(QmlHighlighting::HighlightingMode::Default)
 {
 }
 
@@ -72,10 +72,10 @@ void SemanticTokenFullHandler::process(
         });
         return;
     }
-    auto &&encoded = HighlightingUtils::collectTokens(file, std::nullopt, m_mode);
+    auto &&encoded = QmlHighlighting::Utils::collectTokens(file, std::nullopt, m_mode);
     auto &registeredTokens = m_codeModel->registeredTokens();
     if (!encoded.isEmpty()) {
-        HighlightingUtils::updateResultID(registeredTokens.resultId);
+        QmlHighlighting::Utils::updateResultID(registeredTokens.resultId);
         result = SemanticTokens{ registeredTokens.resultId, encoded };
         registeredTokens.lastTokens = std::move(encoded);
     } else {
@@ -96,7 +96,7 @@ Sends either SemanticTokens or SemanticTokensDelta data as response.
 This is generally requested when the text document is edited after receiving full highlighting data.
 */
 SemanticTokenDeltaHandler::SemanticTokenDeltaHandler(QmlLsp::QQmlCodeModel *codeModel)
-    : QQmlBaseModule(codeModel), m_mode(HighlightingUtils::HighlightingMode::Default)
+    : QQmlBaseModule(codeModel), m_mode(QmlHighlighting::HighlightingMode::Default)
 {
 }
 
@@ -121,17 +121,17 @@ void SemanticTokenDeltaHandler::process(
         });
         return;
     }
-    auto newEncoded = HighlightingUtils::collectTokens(file, std::nullopt, m_mode);
+    auto newEncoded = QmlHighlighting::Utils::collectTokens(file, std::nullopt, m_mode);
     auto &registeredTokens = m_codeModel->registeredTokens();
     const auto lastResultId = registeredTokens.resultId;
-    HighlightingUtils::updateResultID(registeredTokens.resultId);
+    QmlHighlighting::Utils::updateResultID(registeredTokens.resultId);
 
     // Return full token list if result ids not align
     // otherwise compute the delta.
     if (lastResultId == request->m_parameters.previousResultId) {
         result = QLspSpecification::SemanticTokensDelta{
             registeredTokens.resultId,
-            HighlightingUtils::computeDiff(registeredTokens.lastTokens, newEncoded)
+            QmlHighlighting::Utils::computeDiff(registeredTokens.lastTokens, newEncoded)
         };
     } else if (!newEncoded.isEmpty()) {
         result = QLspSpecification::SemanticTokens{ registeredTokens.resultId, newEncoded };
@@ -153,7 +153,7 @@ https://microsoft.github.io/language-server-protocol/specifications/specificatio
 Sends a QLspSpecification::SemanticTokens data as response that is generated for a range of file.
 */
 SemanticTokenRangeHandler::SemanticTokenRangeHandler(QmlLsp::QQmlCodeModel *codeModel)
-    : QQmlBaseModule(codeModel), m_mode(HighlightingUtils::HighlightingMode::Default)
+    : QQmlBaseModule(codeModel), m_mode(QmlHighlighting::HighlightingMode::Default)
 {
 }
 
@@ -183,11 +183,12 @@ void SemanticTokenRangeHandler::process(
     int startOffset =
             int(QQmlLSUtils::textOffsetFrom(code, range.start.line, range.end.character));
     int endOffset = int(QQmlLSUtils::textOffsetFrom(code, range.end.line, range.end.character));
-    auto &&encoded = HighlightingUtils::collectTokens(
-            file, HighlightsRange{ startOffset, endOffset }, m_mode);
     auto &registeredTokens = m_codeModel->registeredTokens();
+    auto &&encoded = QmlHighlighting::Utils::collectTokens(
+            file, QmlHighlighting::HighlightsRange{ startOffset, endOffset }, m_mode);
+
     if (!encoded.isEmpty()) {
-        HighlightingUtils::updateResultID(registeredTokens.resultId);
+        QmlHighlighting::Utils::updateResultID(registeredTokens.resultId);
         result = SemanticTokens{ registeredTokens.resultId, std::move(encoded) };
     } else {
         result = nullptr;
@@ -226,7 +227,7 @@ void QQmlHighlightSupport::setupCapabilities(
 
     if (auto clientInitOptions = clientCapabilities.initializationOptions) {
         if ((*clientInitOptions)[u"qtCreatorHighlighting"_s].toBool(false)) {
-            const auto mode = HighlightingUtils::HighlightingMode::QtCHighlighting;
+            const auto mode = QmlHighlighting::HighlightingMode::QtCHighlighting;
             m_delta.setHighlightingMode(mode);
             m_full.setHighlightingMode(mode);
             m_range.setHighlightingMode(mode);
diff --git a/src/qmlls/qqmlhighlightsupport_p.h b/src/qmlls/qqmlhighlightsupport_p.h
index 80cc036496..f452b88f20 100644
--- a/src/qmlls/qqmlhighlightsupport_p.h
+++ b/src/qmlls/qqmlhighlightsupport_p.h
@@ -56,9 +56,9 @@ public:
     SemanticTokenFullHandler(QmlLsp::QQmlCodeModel *codeModel);
     void process(QQmlBaseModule<SemanticTokensRequest>::RequestPointerArgument req) override;
     void registerHandlers(QLanguageServer *, QLanguageServerProtocol *) override;
-    void setHighlightingMode(HighlightingUtils::HighlightingMode mode) { m_mode = mode; }
+    void setHighlightingMode(QmlHighlighting::HighlightingMode mode) { m_mode = mode; }
     HIDE_UNUSED_OVERRIDES
-    HighlightingUtils::HighlightingMode m_mode;
+    QmlHighlighting::HighlightingMode m_mode;
 };
 
 class SemanticTokenDeltaHandler : public QQmlBaseModule<SemanticTokensDeltaRequest>
@@ -67,9 +67,9 @@ public:
     SemanticTokenDeltaHandler(QmlLsp::QQmlCodeModel *codeModel);
     void process(QQmlBaseModule<SemanticTokensDeltaRequest>::RequestPointerArgument req) override;
     void registerHandlers(QLanguageServer *, QLanguageServerProtocol *) override;
-    void setHighlightingMode(HighlightingUtils::HighlightingMode mode) { m_mode = mode; }
+    void setHighlightingMode(QmlHighlighting::HighlightingMode mode) { m_mode = mode; }
     HIDE_UNUSED_OVERRIDES
-    HighlightingUtils::HighlightingMode m_mode;
+    QmlHighlighting::HighlightingMode m_mode;
 };
 
 class SemanticTokenRangeHandler : public QQmlBaseModule<SemanticTokensRangeRequest>
@@ -78,9 +78,9 @@ public:
     SemanticTokenRangeHandler(QmlLsp::QQmlCodeModel *codeModel);
     void process(QQmlBaseModule<SemanticTokensRangeRequest>::RequestPointerArgument req) override;
     void registerHandlers(QLanguageServer *, QLanguageServerProtocol *) override;
-    void setHighlightingMode(HighlightingUtils::HighlightingMode mode) { m_mode = mode; }
+    void setHighlightingMode(QmlHighlighting::HighlightingMode mode) { m_mode = mode; }
     HIDE_UNUSED_OVERRIDES
-    HighlightingUtils::HighlightingMode m_mode;;
+    QmlHighlighting::HighlightingMode m_mode;;
 };
 
 class QQmlHighlightSupport : public QLanguageServerModule
diff --git a/src/qmlls/qqmlsemantictokens.cpp b/src/qmlls/qqmlsemantictokens.cpp
index 1ad19d7bb0..e15ec2e271 100644
--- a/src/qmlls/qqmlsemantictokens.cpp
+++ b/src/qmlls/qqmlsemantictokens.cpp
@@ -17,7 +17,8 @@ Q_LOGGING_CATEGORY(semanticTokens, "qt.languageserver.semanticTokens")
 using namespace QQmlJS::AST;
 using namespace QQmlJS::Dom;
 using namespace QLspSpecification;
-using namespace HighlightingUtils;
+
+namespace QmlHighlighting {
 
 static int mapToProtocolForQtCreator(QmlHighlightKind highlightKind)
 {
@@ -190,7 +191,7 @@ static std::optional<QmlHighlightKind> resolveJsGlobalObjectKind(const DomItem &
 static int fromQmlModifierKindToLspTokenType(QmlHighlightModifiers highlightModifier)
 {
     using namespace QLspSpecification;
-    using namespace HighlightingUtils;
+    using namespace Utils;
     int modifier = 0;
 
     if (highlightModifier.testFlag(QmlHighlightModifier::QmlPropertyDefinition))
@@ -243,7 +244,7 @@ bool HighlightingVisitor::visitor(Path, const DomItem &item, bool)
         if (!fLocs)
             return true;
         const auto regions = fLocs->info().regions;
-        if (!HighlightingUtils::rangeOverlapsWithSourceLocation(regions[MainRegion],
+        if (!Utils::rangeOverlapsWithSourceLocation(regions[MainRegion],
                                                                 m_range.value()))
             return true;
     }
@@ -312,7 +313,7 @@ void HighlightingVisitor::highlightComment(const DomItem &item)
 {
     const auto comment = item.as<Comment>();
     Q_ASSERT(comment);
-    const auto locs = HighlightingUtils::sourceLocationsFromMultiLineToken(
+    const auto locs = Utils::sourceLocationsFromMultiLineToken(
             comment->info().comment(), comment->info().sourceLocation());
     for (const auto &loc : locs)
         addHighlight(loc, QmlHighlightKind::Comment);
@@ -519,7 +520,7 @@ void HighlightingVisitor::highlightScriptLiteral(const DomItem &item)
         const auto offset = regions[MainRegion].offset;
         const auto length = regions[MainRegion].length;
         const QStringView literalCode = QStringView{code}.mid(offset, length);
-        const auto &locs = HighlightingUtils::sourceLocationsFromMultiLineToken(
+        const auto &locs = Utils::sourceLocationsFromMultiLineToken(
                 literalCode, regions[MainRegion]);
         for (const auto &loc : locs)
             addHighlight(loc, QmlHighlightKind::String);
@@ -838,7 +839,7 @@ void HighlightingVisitor::highlightScriptExpressions(const DomItem &item)
     case DomType::ScriptTemplateStringPart: {
         // handle multiline case
         QString code = item.field(Fields::value).value().toString();
-        const auto &locs = HighlightingUtils::sourceLocationsFromMultiLineToken(
+        const auto &locs = Utils::sourceLocationsFromMultiLineToken(
             code, regions[MainRegion]);
         for (const auto &loc : locs)
             addHighlight(loc, QmlHighlightKind::String);
@@ -860,8 +861,8 @@ This method generates multiple source locations of sub-elements of token split b
 delimiter.
 */
 QList<QQmlJS::SourceLocation>
-HighlightingUtils::sourceLocationsFromMultiLineToken(QStringView stringLiteral,
-                                                     const QQmlJS::SourceLocation &locationInDocument)
+Utils::sourceLocationsFromMultiLineToken(QStringView stringLiteral,
+                                         const QQmlJS::SourceLocation &locationInDocument)
 {
     auto lineBreakLength = qsizetype(std::char_traits<char>::length("\n"));
     const auto lineLengths = [&lineBreakLength](QStringView literal) {
@@ -908,7 +909,7 @@ HighlightingUtils::sourceLocationsFromMultiLineToken(QStringView stringLiteral,
     return result;
 }
 
-QList<int> HighlightingUtils::encodeSemanticTokens(const HighlightsContainer &highlights)
+QList<int> Utils::encodeSemanticTokens(const HighlightsContainer &highlights)
 {
     QList<int> result;
     constexpr auto tokenEncodingLength = 5;
@@ -943,7 +944,7 @@ tokenModifiersList: ["declaration", definition, readonly, static ,,,]
 
 To set "definition" and "readonly", we need to send 0b00000110
 */
-void HighlightingUtils::addModifier(SemanticTokenModifiers modifier, int *baseModifier)
+void Utils::addModifier(SemanticTokenModifiers modifier, int *baseModifier)
 {
     if (!baseModifier)
         return;
@@ -954,7 +955,7 @@ void HighlightingUtils::addModifier(SemanticTokenModifiers modifier, int *baseMo
 \internal
 Check if the ranges overlap by ensuring that one range starts before the other ends
 */
-bool HighlightingUtils::rangeOverlapsWithSourceLocation(const QQmlJS::SourceLocation &loc,
+bool Utils::rangeOverlapsWithSourceLocation(const QQmlJS::SourceLocation &loc,
                                                         const HighlightsRange &r)
 {
     int startOffsetItem = int(loc.offset);
@@ -966,7 +967,7 @@ bool HighlightingUtils::rangeOverlapsWithSourceLocation(const QQmlJS::SourceLoca
 \internal
 Increments the resultID by one.
 */
-void HighlightingUtils::updateResultID(QByteArray &resultID)
+void Utils::updateResultID(QByteArray &resultID)
 {
     int length = resultID.length();
     for (int i = length - 1; i >= 0; --i) {
@@ -986,7 +987,7 @@ A utility method that computes the difference of two list. The first argument is
 of the file before edited. The second argument is the encoded token data after the file is edited. Returns
 a list of SemanticTokensEdit as expected by the protocol.
 */
-QList<SemanticTokensEdit> HighlightingUtils::computeDiff(const QList<int> &oldData, const QList<int> &newData)
+QList<SemanticTokensEdit> Utils::computeDiff(const QList<int> &oldData, const QList<int> &newData)
 {
     // Find the iterators pointing the first mismatch, from the start
     const auto [oldStart, newStart] =
@@ -1026,7 +1027,7 @@ void HighlightingVisitor::addHighlight(const QQmlJS::SourceLocation &loc, QmlHig
         m_highlights.insert(loc.offset,HighlightToken(loc, tokenType, modifierType));
 }
 
-HighlightsContainer HighlightingUtils::visitTokens(const QQmlJS::Dom::DomItem &item,
+HighlightsContainer Utils::visitTokens(const QQmlJS::Dom::DomItem &item,
                                      const std::optional<HighlightsRange> &range,
                                      HighlightingMode mode)
 {
@@ -1035,10 +1036,13 @@ HighlightsContainer HighlightingUtils::visitTokens(const QQmlJS::Dom::DomItem &i
     return highlightDomElements.highlights();
 }
 
-QList<int> HighlightingUtils::collectTokens(const QQmlJS::Dom::DomItem &item,
+QList<int> Utils::collectTokens(const QQmlJS::Dom::DomItem &item,
                                      const std::optional<HighlightsRange> &range,
                                      HighlightingMode mode)
 {
-    return HighlightingUtils::encodeSemanticTokens(visitTokens(item, range, mode));
+    return Utils::encodeSemanticTokens(visitTokens(item, range, mode));
 }
+
+} // namespace QmlHighlighting
+
 QT_END_NAMESPACE
diff --git a/src/qmlls/qqmlsemantictokens_p.h b/src/qmlls/qqmlsemantictokens_p.h
index 854e75bc38..d5c0c6eea2 100644
--- a/src/qmlls/qqmlsemantictokens_p.h
+++ b/src/qmlls/qqmlsemantictokens_p.h
@@ -25,7 +25,7 @@ QT_BEGIN_NAMESPACE
 
 Q_DECLARE_LOGGING_CATEGORY(semanticTokens)
 
-namespace HighlightingUtils {
+namespace QmlHighlighting {
 Q_NAMESPACE
 
 // Protocol agnostic highlighting kinds
@@ -117,8 +117,6 @@ enum class SemanticTokenProtocolTypes {
 };
 Q_ENUM_NS(SemanticTokenProtocolTypes)
 
-} // namespace HighlightingUtils
-
 // Represents a semantic highlighting token
 // startLine and startColumn are 0-based as in LSP spec.
 struct HighlightToken
@@ -155,7 +153,7 @@ struct HighlightToken
 };
 
 using HighlightsContainer = QMap<int, HighlightToken>;
-using QmlHighlightKindToLspKind = int (*)(HighlightingUtils::QmlHighlightKind);
+using QmlHighlightKindToLspKind = int (*)(QmlHighlightKind);
 
 /*!
 \internal
@@ -167,7 +165,7 @@ struct HighlightsRange
     int endOffset;
 };
 
-namespace HighlightingUtils
+namespace Utils
 {
     QList<int> encodeSemanticTokens(const HighlightsContainer &highlights);
     QList<QQmlJS::SourceLocation>
@@ -183,15 +181,14 @@ namespace HighlightingUtils
     HighlightsContainer visitTokens(const QQmlJS::Dom::DomItem &item,
                                  const std::optional<HighlightsRange> &range,
                                  HighlightingMode mode = HighlightingMode::Default);
-} // namespace HighlightingUtils
+} // namespace Utils
 
 class HighlightingVisitor
 {
 public:
     HighlightingVisitor(const QQmlJS::Dom::DomItem &item,
                         const std::optional<HighlightsRange> &range,
-                        HighlightingUtils::HighlightingMode mode =
-                                HighlightingUtils::HighlightingMode::Default);
+                        HighlightingMode mode = HighlightingMode::Default);
     const HighlightsContainer &hightights() const { return m_highlights; }
     HighlightsContainer &highlights() { return m_highlights; }
 
@@ -213,15 +210,16 @@ private:
     void highlightScriptExpressions(const QQmlJS::Dom::DomItem &item);
     void highlightCallExpression(const QQmlJS::Dom::DomItem &item);
     void highlightFieldMemberAccess(const QQmlJS::Dom::DomItem &item, QQmlJS::SourceLocation loc);
-    void addHighlight(const QQmlJS::SourceLocation &loc, HighlightingUtils::QmlHighlightKind,
-                      HighlightingUtils::QmlHighlightModifiers =
-                              HighlightingUtils::QmlHighlightModifier::None);
+    void addHighlight(const QQmlJS::SourceLocation &loc, QmlHighlightKind,
+                      QmlHighlightModifiers = QmlHighlightModifier::None);
 private:
     HighlightsContainer m_highlights;
     std::optional<HighlightsRange> m_range;
     QmlHighlightKindToLspKind m_mapToProtocol;
 };
 
+} // namespace QmlHighlighting
+
 QT_END_NAMESPACE
 
 #endif // QQMLSEMANTICTOKENS_P_H
diff --git a/tests/auto/qmlls/modules/tst_qmlls_modules.cpp b/tests/auto/qmlls/modules/tst_qmlls_modules.cpp
index b075c29cfe..1202bbfbd3 100644
--- a/tests/auto/qmlls/modules/tst_qmlls_modules.cpp
+++ b/tests/auto/qmlls/modules/tst_qmlls_modules.cpp
@@ -1791,8 +1791,8 @@ void tst_qmlls_modules::semanticHighlightingFull()
 {
     QFETCH(QString, filePath);
     const auto item = fileObject(testFile(filePath));
-    HighlightsContainer highlights;
-    const auto expectedData = HighlightingUtils::collectTokens(item, std::nullopt);
+    QmlHighlighting::HighlightsContainer highlights;
+    const auto expectedData = QmlHighlighting::Utils::collectTokens(item, std::nullopt);
 
     const auto uri = openFile(filePath);
     QVERIFY(uri);
@@ -1834,12 +1834,13 @@ void tst_qmlls_modules::semanticHighlightingRange()
     QFETCH(QLspSpecification::Range, range);
 
     const auto item = fileObject(testFile(filePath));
-    HighlightsContainer highlights;
+    QmlHighlighting::HighlightsContainer highlights;
     const auto qmlFile = item.as<QQmlJS::Dom::QmlFile>();
     const auto code = qmlFile->code();
     const int startOffset = int(QQmlLSUtils::textOffsetFrom(code, range.start.line, range.end.character));
     const int endOffset = int(QQmlLSUtils::textOffsetFrom(code, range.end.line, range.end.character));
-    const auto expectedData = HighlightingUtils::collectTokens(item, HighlightsRange{startOffset, endOffset});
+    const auto expectedData = QmlHighlighting::Utils::collectTokens(
+            item, QmlHighlighting::HighlightsRange{ startOffset, endOffset });
 
     const auto uri = openFile(filePath);
     QVERIFY(uri);
@@ -1884,9 +1885,10 @@ void tst_qmlls_modules::semanticHighlightingDelta()
 
     const auto fileItem = fileObject(testFile(filePath));
     const auto deltaFileItem = fileObject(testFile(deltaFilePath));
-    auto fullDocumentSemanticTokensData = HighlightingUtils::collectTokens(fileItem, std::nullopt);
-    auto editedDocumentSemanticTokensData = HighlightingUtils::collectTokens(deltaFileItem, std::nullopt);
-    const auto expectedEdits = HighlightingUtils::computeDiff(fullDocumentSemanticTokensData, editedDocumentSemanticTokensData);
+    auto fullDocumentSemanticTokensData = QmlHighlighting::Utils::collectTokens(fileItem, std::nullopt);
+    auto editedDocumentSemanticTokensData = QmlHighlighting::Utils::collectTokens(deltaFileItem, std::nullopt);
+    const auto expectedEdits = QmlHighlighting::Utils::computeDiff(
+            fullDocumentSemanticTokensData, editedDocumentSemanticTokensData);
 
     const auto uri = openFile(filePath);
     QVERIFY(uri);
diff --git a/tests/auto/qmlls/utils/tst_qmlls_highlighting.cpp b/tests/auto/qmlls/utils/tst_qmlls_highlighting.cpp
index db1ae2e0ff..231dc4fd23 100644
--- a/tests/auto/qmlls/utils/tst_qmlls_highlighting.cpp
+++ b/tests/auto/qmlls/utils/tst_qmlls_highlighting.cpp
@@ -15,6 +15,7 @@
 #include <qlist.h>
 
 using namespace QLspSpecification;
+using namespace QmlHighlighting;
 
 tst_qmlls_highlighting::tst_qmlls_highlighting()
     : QQmlDataTest(QT_QMLLS_HIGHLIGHTS_DATADIR) , m_highlightingDataDir(QT_QMLLS_HIGHLIGHTS_DATADIR + "/highlights"_L1)
@@ -73,7 +74,7 @@ void tst_qmlls_highlighting::encodeSemanticTokens()
 {
     QFETCH(HighlightsContainer, highlights);
     QFETCH(QList<int>, expectedMemoryLayout);
-    const auto encoded = HighlightingUtils::encodeSemanticTokens(highlights);
+    const auto encoded = Utils::encodeSemanticTokens(highlights);
     QCOMPARE(encoded, expectedMemoryLayout);
 }
 
@@ -116,7 +117,7 @@ void tst_qmlls_highlighting::sourceLocationsFromMultiLineToken()
 
     auto *literal = QQmlJS::AST::cast<QQmlJS::AST::StringLiteral *>(expression);
     const auto locs =
-            HighlightingUtils::sourceLocationsFromMultiLineToken(source, literal->literalToken);
+            Utils::sourceLocationsFromMultiLineToken(source, literal->literalToken);
 
     [&]() {
         QCOMPARE(locs.size(), expectedLines.size());
@@ -145,7 +146,7 @@ void tst_qmlls_highlighting::sourceLocationsFromMultiLineToken()
 void tst_qmlls_highlighting::highlights_data()
 {
     using namespace QQmlJS::Dom;
-    using namespace HighlightingUtils;
+    using namespace Utils;
     QTest::addColumn<DomItem>("fileItem");
     QTest::addColumn<HighlightToken>("expectedHighlightedToken");
 
@@ -788,7 +789,7 @@ void tst_qmlls_highlighting::highlights()
     QFETCH(DomItem, fileItem);
     QFETCH(HighlightToken, expectedHighlightedToken);
 
-    const auto highlights = HighlightingUtils::visitTokens(fileItem,std::nullopt);
+    const auto highlights = Utils::visitTokens(fileItem,std::nullopt);
 
     [&]() {
         QVERIFY(highlights.contains(expectedHighlightedToken.offset));
@@ -859,7 +860,7 @@ void tst_qmlls_highlighting::rangeOverlapsWithSourceLocation()
     QFETCH(QQmlJS::SourceLocation, sourceLocation);
     QFETCH(HighlightsRange, range);
     QFETCH(bool, overlaps);
-    QVERIFY(overlaps == HighlightingUtils::rangeOverlapsWithSourceLocation(sourceLocation, range));
+    QVERIFY(overlaps == Utils::rangeOverlapsWithSourceLocation(sourceLocation, range));
 }
 
 void tst_qmlls_highlighting::updateResultID_data()
@@ -878,7 +879,7 @@ void tst_qmlls_highlighting::updateResultID()
     QFETCH(QByteArray, currentId);
     QFETCH(QByteArray, expectedNextId);
 
-    HighlightingUtils::updateResultID(currentId);
+    Utils::updateResultID(currentId);
     QCOMPARE(currentId, expectedNextId);
 }
 
@@ -959,7 +960,7 @@ void tst_qmlls_highlighting::computeDiff()
     QFETCH(QList<int>, newData);
     QFETCH(QList<SemanticTokensEdit>, expected);
 
-    const auto edits = HighlightingUtils::computeDiff(oldData, newData);
+    const auto edits = Utils::computeDiff(oldData, newData);
     QCOMPARE(edits.size(), expected.size());
 
     qsizetype i = 0;
-- 
2.51.2

